packages/wrangler/src/metrics/metrics-dispatcher.ts

```diff
@@ -19,62 +56,105 @@ export async function getMetricsDispatcher(options: MetricsConfigOptions) {
 		 *  - additional properties are camelCased
 		 */
 		async sendEvent(name: string, properties: Properties = {}): Promise<void> {
-			await dispatch({ type: "event", name, properties });
+			await dispatch({
+				name,
+				properties: {
+					category: "Workers",
+					wranglerVersion,
+					os: getOS(),
+					...properties,
+				},
+			});
 		},
 
 		/**
-		 * Dispatch a user profile information to the analytics target.
+		 * Dispatches `wrangler command started / completed / errored` events
 		 *
-		 * This call can be used to inform the analytics target of relevant properties associated
-		 * with the current user.
+		 * This happens on every command execution, and will (hopefully) replace sendEvent soon.
+		 * However to prevent disruption, we're adding under `sendNewEvent` for now.
 		 */
-		async identify(properties: Properties): Promise<void> {
-			await dispatch({ type: "identify", name: "identify", properties });
+		async sendNewEvent<EventName extends Events["name"]>(
+			name: EventName,
+			properties: Omit<
+				Extract<Events, { name: EventName }>["properties"],
+				keyof CommonEventProperties
+			>
+		): Promise<void> {
+			if (
+				properties.command === "wrangler telemetry disable" ||
+				properties.command === "wrangler metrics disable"
+			) {
+				return;
+			}
+			printMetricsBanner();
+			const argsUsed = sanitiseUserInput(properties.args ?? {});
+			const argsCombination = argsUsed.sort().join(", ");
+			const commonEventProperties: CommonEventProperties = {
+				amplitude_session_id,
+				amplitude_event_id: amplitude_event_id++,
+				wranglerVersion,
+				osPlatform,
+				osVersion,
+				nodeVersion,
+				packageManager,
+				isFirstUsage,
+				configFileType,
+				isCI,
+				isInteractive,
+				argsUsed,
+				argsCombination,
```

Comment:
To me it feels like this would be clearer to read if the helper functions were inline here e.g. `isCI: CI.isCI()`. In most cases they're a single function, the current layout is quite verbose & requires some funky renaming (like `_IsInteractive()`